ledger-api/grpc-definitions/com/daml/ledger/api/v1/transaction_filter.proto

```diff
@@ -12,30 +12,102 @@ option java_outer_classname = "TransactionFilterOuterClass";
 option java_package = "com.daml.ledger.api.v1";
 option csharp_namespace = "Com.Daml.Ledger.Api.V1";
 
-// Used for filtering Transaction and Active Contract Set streams.
-// Determines which on-ledger events will be served to the client.
+
+// Filtering Transaction, TransactionTree, and Active Contract Set streams
+//////////////////////////////////////////////////////////////////////////
+
+// A filter both for filtering create and archive events; and for filtering
+// transaction trees.
 message TransactionFilter {
 
-  // Keys of the map determine which parties' on-ledger transactions are being queried.
-  // Values of the map determine which events are disclosed in the stream per party.
-  // At the minimum, a party needs to set an empty Filters message to receive any events.
   // Each key must be a valid PartyIdString (as described in ``value.proto``).
+  // For transaction trees only the listed parties matter, and all subtrees
+  // whose root is has one of the listed parties as an informee are returned.
+  // Create and and archive events are returned for all contracts whose
```

Comment:
Transaction trees also contain create events. Maybe use a clearer separation?
```
// The interpretation of the filter depends on the stream being filtered:
// - Transaction tree streams: Only the listed parties matter, and all subtrees
//    whose root is has one of the listed parties as an informee are returned.
// - Transaction and ACS streams: Create and archive events are returned for all contracts whose
//    stakeholders include at least one of the listed parties and match the
//    per-party filter.
```

Should we document that it's an error to include a non-wildcard filter for a transaction tree stream?

Reply:
Good suggestion. Added.